terraform/environments/data-platform-apps-and-tools/secrets.tf

```diff
@@ -3,6 +3,31 @@ data "aws_ssm_parameter" "modernisation_platform_account_id" {
   name = "modernisation_platform_account_id"
 }
 
+# Create a new secret in AWS SecretsManager for Gov.UK Notify API key
+resource "aws_secretsmanager_secret" "govuk_notify_api_key" {
+  count = terraform.workspace == "data-platform-apps-and-tools-production" ? 1 : 0
+
+  name = "gov-uk-notify/production/api-key"
+}
+
+# Add the Gov.UK Notify API key to the secret
```

Comment:
I think we would want to cut the below block where we attempt to populate with a value, otherwise the actual value we wish to set would be overwritten every time someone merges to this codebase